Modules/itertoolsmodule.c

```diff
@@ -213,19 +219,31 @@ batched_next(batchedobject *bo)
     if (PyErr_Occurred()) {
         if (!PyErr_ExceptionMatches(PyExc_StopIteration)) {
             /* Input raised an exception other than StopIteration */
+#ifdef Py_GIL_DISABLED
```

Comment:
No need to check for `Py_GIL_DISABLED`, FT atomics do that internally and assigning -1 even in gil builds is of no harm.

Reply:
The assignment does no harm, but it is a bit of double work, which is the reason I avoided it earlier. The double work (assign -1 and clear the iterator) is only done on error or exhausting the iterators though (not in the hot path), so it is fine.